### PR TITLE
Prepare 0.21.2

### PR DESCRIPTION
**Pull request by daxpedda**
- [x] Update changelog.
- [x] `cargo update`.
- [x] `cargo outdated`.
- [x] `cargo test --all-features`.
- [x] Update `version` in `Cargo.toml`.
- [x] `cargo publish --dry-run`
 
Happy to take any advice on the changelog.

### CODE REVIEW

**Review by ctz — APPROVED** [review]
Thanks, this looks good to me.

Just for my understanding:

>  `RootCertStore::add_parsable_certificates` now takes a `&[impl AsRef<[u8]>]`.

This isn't a breaking change, because previously it took `&[Vec<u8>]` and [generalisation of function arguments is allowed](https://doc.rust-lang.org/cargo/reference/semver.html#minor-generalizing-a-function-to-use-generics-supporting-original-type).

**Comment by djc** [comment]
> Just for my understanding:
> 
> > `RootCertStore::add_parsable_certificates` now takes a `&[impl AsRef<[u8]>]`.
> 
> This isn't a breaking change, because previously it took `&[Vec<u8>]` and [generalisation of function arguments is allowed](https://doc.rust-lang.org/cargo/reference/semver.html#minor-generalizing-a-function-to-use-generics-supporting-original-type).

Yup, exactly.

File: README.md [hunk 1]
```diff
@@ -20,8 +20,14 @@ If you'd like to help out, please see [CONTRIBUTING.md](CONTRIBUTING.md).
 
 ## Release history
 
-* Next release:
+* Release 0.21.2 (2023-06-14)
   - Bump MSRV to 1.60 to track similar change in dependencies.
+  - Differentiate between unexpected and expected EOF in `Stream` and `OwnedStream`.
+  - `RootCertStore::add_parsable_certificates` now takes a `&[impl AsRef<[u8]>]`.
+  - Add QUIC V2 support.
+  - Upgrade `env_logger` to v0.10.
```
Comment (djc) [inline README.md:28]: IIRC env_logger is a dev-dep only, so probably doesn't need to go in the changelog.

File: README.md [hunk 2]
```diff
@@ -20,8 +20,14 @@ If you'd like to help out, please see [CONTRIBUTING.md](CONTRIBUTING.md).
...
+  - Correctly provide an empty session ID if none is available for TLS 1.2.
+  - Bug fix: abort clients offering "pre_shared_key" without a "psk_key_exchange_modes" extension.
```
Comment (djc) [inline README.md:30]: Historically we haven't really made node of minor bugfixes like this in the changelog (I would probably only keep the QUIC V2 support and the `Stream` EOF handling), @ctz do you feel we should change that?

File: README.md (hunk 1 above)
Comment (daxpedda) [reply README.md:28]: Ah, right, my bad.

File: README.md (hunk 2 above)
Comment (ctz) [reply README.md:29]: Yeah agree that it's a minor bug fix (and never going to be seen in the wild). I probably wouldn't put it in a change log (cos it's not interesting or actionable for users).

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Published and pushed tag.

**Pull request merged**